### ISSUE TITLE
Vim: Caught deadly signal ABRT Vim: Finished. Abort trap: 6

### ISSUE THREAD

**Issue by Bensji** [wincent/command-t#324]
After installing vim with homebrew, command-t doesn't work anymore.

This is the output I get:

`Vim: Caught deadly signal ABRT
Vim: Finished.
Abort trap: 6`

Not sure if this is related to [another topic I found](https://github.com/Valloric/YouCompleteMe/issues/8) about the version of Python?

Because I already tried to unlink python, install vim again with brew and try to rebuild command-T (doing   cd ~/.vim/bundle/command-t/ruby/command-t/ext/command-t
  ruby extconf.rb
  make), but that didn't work.

` $ otool -L `which vim`
/usr/local/bin/vim:
	/usr/lib/libncurses.5.4.dylib (compatibility version 5.4.0, current version 5.4.0)
	/usr/lib/libiconv.2.dylib (compatibility version 7.0.0, current version 7.0.0)
	/System/Library/Frameworks/AppKit.framework/Versions/C/AppKit (compatibility version 45.0.0, current version 1504.83.101)
	/usr/local/opt/perl/lib/perl5/5.26.1/darwin-thread-multi-2level/CORE/libperl.dylib (compatibility version 5.26.0, current version 5.26.1)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1238.60.2)
	/usr/lib/libutil.dylib (compatibility version 1.0.0, current version 1.0.0)
	/usr/local/opt/python/Frameworks/Python.framework/Versions/2.7/Python (compatibility version 2.7.0, current version 2.7.0)
	/usr/local/opt/ruby/lib/libruby.2.5.dylib (compatibility version 2.5.0, current version 2.5.0)
	/usr/lib/libobjc.A.dylib (compatibility version 1.0.0, current version 228.0.0)
	/System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation (compatibility version 150.0.0, current version 1349.90.0)
	/System/Library/Frameworks/CoreServices.framework/Versions/A/CoreServices (compatibility version 1.0.0, current version 775.19.0)
	/System/Library/Frameworks/Foundation.framework/Versions/C/Foundation (compatibility version 300.0.0, current version 1349.91.0)`

System:
macOSX Sierra (10.12.6)

vim --version
`$ vim --version
VIM - Vi IMproved 8.0 (2016 Sep 12, compiled Jan  8 2018 13:56:05)
macOS version
Included patches: 1-1400
Compiled by Homebrew
Huge version without GUI.  Features included (+) or not (-):
+acl               +farsi             +mouse_sgr         -tag_any_white
+arabic            +file_in_path      -mouse_sysmouse    -tcl
+autocmd           +find_in_path      +mouse_urxvt       +termguicolors
-autoservername    +float             +mouse_xterm       +terminal
-balloon_eval      +folding           +multi_byte        +terminfo
+balloon_eval_term -footer            +multi_lang        +termresponse
-browse            +fork()            -mzscheme          +textobjects
++builtin_terms    -gettext           +netbeans_intg     +timers
+byte_offset       -hangul_input      +num64             +title
+channel           +iconv             +packages          -toolbar
+cindent           +insert_expand     +path_extra        +user_commands
-clientserver      +job               +perl              +vertsplit
+clipboard         +jumplist          +persistent_undo   +virtualedit
+cmdline_compl     +keymap            +postscript        +visual
+cmdline_hist      +lambda            +printer           +visualextra
+cmdline_info      +langmap           +profile           +viminfo
+comments          +libcall           +python            +vreplace
+conceal           +linebreak         -python3           +wildignore
+cryptv            +lispindent        +quickfix          +wildmenu
+cscope            +listcmds          +reltime           +windows
+cursorbind        +localmap          +rightleft         +writebackup
+cursorshape       -lua               +ruby              -X11
+dialog_con        +menu              +scrollbind        -xfontset
+diff              +mksession         +signs             -xim
+digraphs          +modify_fname      +smartindent       -xpm
-dnd               +mouse             +startuptime       -xsmp
-ebcdic            -mouseshape        +statusline        -xterm_clipboard
+emacs_tags        +mouse_dec         -sun_workshop      -xterm_save
+eval              -mouse_gpm         +syntax
+ex_extra          -mouse_jsbterm     +tag_binary
+extra_search      +mouse_netterm     +tag_old_static
   system vimrc file: "$VIM/vimrc"
     user vimrc file: "$HOME/.vimrc"
 2nd user vimrc file: "~/.vim/vimrc"
      user exrc file: "$HOME/.exrc"
       defaults file: "$VIMRUNTIME/defaults.vim"
  fall-back for $VIM: "/usr/local/share/vim"
Compilation: clang -c -I. -Iproto -DHAVE_CONFIG_H   -DMACOS_X -DMACOS_X_DARWIN  -g -O2 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1
Linking: clang   -L. -fstack-protector -L/usr/local/lib -L/usr/local/opt/libyaml/lib -L/usr/local/opt/openssl/lib -L/usr/local/opt/readline/lib  -L/usr/local/lib -o vim        -lncurses -liconv -framework AppKit   -mmacosx-version-min=10.12 -fstack-protector-strong -L/usr/local/lib  -L/usr/local/Cellar/perl/5.26.1/lib/perl5/5.26.1/darwin-thread-multi-2level/CORE -lperl -lm -lutil -lc -F/usr/local/opt/python/Frameworks -framework Python   -lruby.2.5.0 -lobjc`

Thanks!

**Comment by wincent** [comment]
Python has nothing to do with Command-T so you can ignore that. I am on my phone now but take a look at the troubleshooting section in the docs or at the closed issues here: the key is to compile Command-T with the same version of Ruby that Vim is linked against.

**Comment by joshdeese** [comment]
I had a similar issue. I got this error after updating vim/ruby. What worked for me was:

    cd ~/.vim/bundle/command-t/ruby/command-t/ext/command-t
    make clean
    ruby extconf.rb
    make

**Comment by stepchud** [comment]
I get this same error when compiling vim from scratch (which was only necessary b/c the ruby/vim combination on Mojave doesn't want to work with command-t but I digress...)
Something about the `make install` step causes this SIGABRT with command-t. Yes I've tried rebuilding command-t using make clean; ruby extconf.rb; make. The version of vim I built in my vim/src/ folder **works with command-t**. It is different from what gets copied to /usr/local/bin during make install. The only thing I see is it runs `strip /usr/local/bin/vim` after copy, but I don't know why that would cause SIGABRT. I'm falling back to copying the vim manually into /usr/local/bin. Seems like I'm back in a working state now. 🤷‍♂️
